packages/use-i18n/src/usei18n.tsx

```diff
@@ -99,19 +108,23 @@ interface Context<Locale extends BaseLocale> {
 // useI18n / useTranslation requires to explicitely give a Locale to use.
 const I18nContext = createContext<Context<any> | undefined>(undefined)
 
-export function useI18n<Locale extends BaseLocale>(): Context<Locale> {
+export function useI18n<
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  Locale extends BaseLocale = {},
+>(): RequiredGenericContext<Locale> {
   const context = useContext(I18nContext)
   if (context === undefined) {
     throw new Error('useI18n must be used within a I18nProvider')
   }
 
-  return context as unknown as Context<Locale>
+  return context as unknown as RequiredGenericContext<Locale>
 }
 
-export function useTranslation<Locale extends BaseLocale>(
+// eslint-disable-next-line @typescript-eslint/ban-types
```

Comment:
which type is banned ?

Reply:
`{}`

Reply:
Why not use object ? Or Record ?

Reply:
Because it has to be an `{}` to check if it's empty (= `keyof Locale extends never`) or not in order to show the error message. This [this playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDWEQHsBmUDeBfAUFFAPTFQB6A-PvqJFAEILKqZQDyARgFYQDGwQiTJUhtcNADCzFOiwAlfmgBOAEwA8AZ2DKAlgDsA5gBoo+iADcIygHxDSFSkA)

Reply:
<img width="259" alt="Screenshot 2022-10-26 at 11 49 56" src="https://user-images.githubusercontent.com/2802867/197995070-f7659886-f4dc-4052-a9bd-c827f943a1f3.png">

Reply:
`object` by itself doesn't satisfy the constraint of `BaseLocale`, because it can be any object but we require an object with keys being strings.